layer.py

```diff
@@ -88,11 +88,19 @@ def __init__(self):
 
 
 class LayerSource(object):
+    ATTACHMENT_EXPRESSIONS = [
+        "'files/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '_#~filename~#'",  # Files
+        "'DCIM/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '.#~extension~#'",  # Image
+        "",  # Web
+        "'audio/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '.#~extension~#'",  # Audio
+        "'video/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '.#~extension~#'",  # Video
+    ]
```

Comment:
Actually a second though, can we make this a disctionary, so we have something like:

```
class AtachmentType(Enum):
	FILE = 0
        IMAGE = 1
       	WEB = 2
       	AUDIO = 3
       	VIDEO = 4

ATTACHMENT_EXPRESSIONS = {
        AttachmentType.FILE: "'files/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '_#~filename~#'",
        AttachmentType.IMAGE: "'DCIM/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '.#~extension~#'",
        AttachmentType.WEB: "",
        AttachmentType.AUDIO: "'audio/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '.#~extension~#'",
        AttachmentType.VIDEO: "'video/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '.#~extension~#'",
}
```

Reply:
Greatly improves readability and reduces these magic indeces floating around.